tests/components/sensor/test_awair.py

```diff
@@ -0,0 +1,281 @@
+"""Tests for the Awair sensor platform."""
+
+from contextlib import contextmanager
+from datetime import timedelta
+import json
+import logging
+from unittest.mock import patch
+
+from homeassistant.components.sensor import DOMAIN as SENSOR_DOMAIN
+from homeassistant.components.sensor.awair import (
+    ATTR_LAST_API_UPDATE, ATTR_TIMESTAMP, DEVICE_CLASS_CARBON_DIOXIDE,
+    DEVICE_CLASS_PM2_5, DEVICE_CLASS_SCORE,
+    DEVICE_CLASS_VOLATILE_ORGANIC_COMPOUNDS)
+from homeassistant.const import (
+    DEVICE_CLASS_HUMIDITY, DEVICE_CLASS_TEMPERATURE, STATE_UNAVAILABLE,
+    TEMP_CELSIUS)
+from homeassistant.setup import async_setup_component
+from homeassistant.util.dt import parse_datetime, utcnow
+
+from tests.common import async_fire_time_changed, load_fixture, mock_coro
+
+DISCOVERY_CONFIG = {
+    'sensor': {
+        'platform': 'awair',
+        'access_token': 'qwerty',
+    }
+}
+
+MANUAL_CONFIG = {
+    'sensor': {
+        'platform': 'awair',
+        'access_token': 'qwerty',
+        'devices': [
+            {'uuid': 'awair_foo'}
+        ]
+    }
+}
+
+_LOGGER = logging.getLogger(__name__)
+
+NOW = utcnow()
+AIR_DATA_FIXTURE = json.loads(load_fixture('awair_air_data_latest.json'))
+AIR_DATA_FIXTURE[0][ATTR_TIMESTAMP] = str(NOW)
+AIR_DATA_FIXTURE_UPDATED = json.loads(
+    load_fixture('awair_air_data_latest_updated.json'))
+AIR_DATA_FIXTURE_UPDATED[0][ATTR_TIMESTAMP] = str(NOW + timedelta(minutes=5))
+
+
+@contextmanager
+def alter_time(retval):
+    """Manage multiple time mocks."""
+    patch_one = patch('homeassistant.util.dt.utcnow', return_value=retval)
+    patch_two = patch('homeassistant.util.utcnow', return_value=retval)
+
+    with patch_one, patch_two:
+        yield
+
+
+async def setup_awair(hass, config=None):
+    """Load the Awair platform."""
+    devices_json = json.loads(load_fixture('awair_devices.json'))
+    devices_mock = mock_coro(devices_json)
+    devices_patch = patch('python_awair.AwairClient.devices',
+                          return_value=devices_mock)
+    air_data_mock = mock_coro(AIR_DATA_FIXTURE)
+    air_data_patch = patch('python_awair.AwairClient.air_data_latest',
+                           return_value=air_data_mock)
+
+    if config is None:
+        config = DISCOVERY_CONFIG
+
+    with devices_patch, air_data_patch, alter_time(NOW):
+        assert await async_setup_component(hass, SENSOR_DOMAIN, config)
+
+    await hass.async_block_till_done()
```

Comment:
Indent this under the context manager scope.

Reply:
:+1: